Sources/PostgresNIO/New/Connection State Machine/ConnectionStateMachine.swift

```diff
@@ -113,17 +113,20 @@ struct ConnectionStateMachine {
     }
     
     private var state: State
+    private let requireBackendKeyData: Bool
     private var taskQueue = CircularBuffer<PSQLTask>()
     private var quiescingState: QuiescingState = .notQuiescing
     
-    init() {
+    init(requireBackendKeyData: Bool = true) {
```

Comment:
We should not default this property here. Whenever we use the state machine we should be explicit.